lib/Template/Directive.pm

```diff
@@ -429,10 +429,11 @@ sub if {
 #------------------------------------------------------------------------
 
 sub foreach {
-    my ($self, $target, $list, $args, $block, $label) = @_;
+    my ($self, $target, $list, $args, $block, $else, $label) = @_;
```

Comment:
when I read this, I've the feeling that this going to break previous behaviors where foreach was called with a label, as the label is going to be shift in else statement.

One suggestion would be to move the else as last argument of this function and use an extra helper in the parser that would wrap this function